### PR TITLE
Add WithHeader option

### PR DESCRIPTION
**Pull request by benpate**
Addresses Issue #5

### CODE REVIEW

**Comment by benpate** [comment]
I saw this project on Reddit, and the timing is perfect because I need something exactly like this -- but with support for HTTP headers.  I'm pretty new with GitHub pull requests, so please excuse me if I've missed anything.  

I had a little trouble running the tests built into the package, but my tests against my own server are working just fine.  If you've got some guidance there, I'll make any changes necessary.  Hopefully this checks out with the way you'd implement this on your own.

Best wishes!

**Comment by BrandonRomano** [comment]
@benpate I'll check this out in the morning!  From a quick glance this looks good, but I'd like to give it a more thorough look + test.

File: wrecker.go [hunk 1]
```diff
@@ -66,6 +67,11 @@ func (w *Wrecker) sendRequest(verb string, requestURL string, bodyParams url.Val
 	}
 	clientReq.Header.Add("Content-Type", w.DefaultContentType)
 
+	// Add headers to clientReq
+	for index, value := range headers {
```
Comment (BrandonRomano) [inline wrecker.go:71]: Could this possibly be `key, value` as opposed to `index, value`?

**Comment by BrandonRomano** [comment]
Hey @benpate -- I actually decided to check this out now.

This looks good + seems to be working!

One thing though -- The updated tests actually don't prove that this is working (even though I'm very confident it is)!

What I think would be best (for now) would be to update let's say the [Delete request](https://github.com/BrandonRomano/wrecker/blob/master/test/server.go#L89) in the test server to require some specific header, else it would fail the request.

Then in the test file we can send that header, and then check if the request succeeded (thus proving a header was successfully sent).

I could take a stab at this tomorrow morning (unless you want to handle it)!

**Comment by BrandonRomano** [comment]
Hey @benpate.  I'm going to merge this in and then improve the tests right after.

Thanks for the contribution :+1:

**Pull request merged**